compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/NativeInputEventsProcessor.kt

```diff
@@ -146,6 +148,8 @@ internal abstract class NativeInputEventsProcessor(
                 val deleteSize = deleteContentBackwardSize
                 if (deleteSize > 0) {
                     add(DeleteSurroundingTextCommand(deleteSize, 0))
+                } else if (deleteSize == 0) {
+                    add(BackspaceCommand())
```

Comment:
As I remember from the logs you shared from iPad, only `deleteContentBackward` was dispatched there for this type of input.

Is it necessary to add this check (`deleteSize == 0`) here too? 

The name of the events suggests that `deleteContentBackward` can indeed have no selection - so just delete like a backspace would do.

But `insertReplacementText` hints that a selection must be present to replace something.

Reply:
good catch! - this is fallacy of tuning something to test - (in this case, for `SafariCompositeInput::`input hangul-hol`) let me fix test instead